msvc/scripts/genproj.cs

```diff
@@ -887,6 +945,26 @@ public VsCsproj Generate (string library_output, Dictionary<string,MsbuildGenera
 
 		bool basic_or_build = (library.Contains ("-basic") || library.Contains ("-build"));
 
+		// If an EXE is built with nostdlib, it won't work unless run with mono.exe. This stops our build steps
+		//  from working in visual studio (because we already replace @MONO@ with '' on Windows.)
+
+		if (Target != Target.Library)
+			StdLib = true;
+
+		// We have our target framework set to 4.5 in many places because broken scripts check for files with 4.5
+		//  in the path, even though we compile code that uses 4.6 features. So we need to manually fix that here.
```

Comment:
Where is that?

Reply:
Zoltan and Jo gave me the impression it needs to remain 4.5 in the build files forever because we use 4_5 as an identifier for the desktop profile. I don't really know what would break.

Reply:
I still don't understand this, where is this used ?

Reply:
If I don't change it to 4.6.2 some bits fail to compile because we use newer APIs. If you want I can try changing it in the source makefiles instead?